src/dbus_fast/aio/proxy_object.py

```diff
@@ -87,7 +87,11 @@ async def method_fn(*args, flags=MessageFlag.NONE):
                     member=intr_method.name,
                     signature=intr_method.in_signature,
                     body=input_body,
-                    flags=flags,
+                    flags=(
+                        flags ^ MessageFlag.REMOVE_SIGNATURE
+                        if flags & MessageFlag.REMOVE_SIGNATURE
+                        else flags
+                    ),
```

Comment:
```suggestion
                    flags=flags & ~MessageFlag.REMOVE_SIGNATURE,
```

Reply:
Or instead of silently fixing it, throw a ValueError?

Reply:
It doesn't fix it actually. It's totally fine to pass in `REMOVE_SIGNATURE` to this method. The reason its doing this is in case we ever want to support the `REMOVE_SIGNATURE` flag in the underlying `MessageBus.call` method. If the signature is stripped by the call method method then this method (in the proxy object) will break. As the code that follows relies on checking the signature.

Basically I just want to be able to handle removing the variants/signature data in this code and not have `call` ever strip it out.

Reply:
Also that is a way better way to do this, will change it 👍